packages/components/src/Text/src/Text.tsx

```diff
@@ -0,0 +1,68 @@
+import { type StyledComponentProps, useStyledSystem, type ResponsiveProp, useResponsiveValue } from "@hopper-ui/styled-system";
+import { type ForwardedRef, forwardRef, type CSSProperties } from "react";
+import { Text as RACText, useContextProps, type TextProps as RACTextProps } from "react-aria-components";
+import clsx from "clsx";
+import styles from "./Text.module.css";
+import { cssModule } from "../../utils/src/css-module.ts";
+import { TextContext } from "./TextContext.ts";
+
+export const GlobalTextCssSelector = "hop-Text";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultTextSlot = "text";
+
+export type RACTextPropsToOmit = "elementType";
+
+export interface TextProps extends StyledComponentProps<Omit<RACTextProps, RACTextPropsToOmit>> {
+    /**
+     * The Typography Type Scale to use.
+     * @default "md"
+     */
+    size?: ResponsiveProp<"inherit" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl">;
```

Comment:
Should the Text component support variations? Will we have another component for headings?
Where would overline and accents be included?

Reply:
We will have other components for Headings and Overline yes